keyboards/nasu/keymaps/mariocs/keymap.c

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2021 MarioCadenas
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    switch (get_highest_layer(state)) {
+    case 1:
+        setPinInputHigh(B0);
+        break;
+    default:
+        setPinInputLow(B0);
+        break;
+    }
+  return state;
+}
```

Comment:
This may be a better way of handling this: 

```suggestion
void keyboard_post_init_user(void) {
    setPinInputHigh(B0);
    writePinHigh(B0);
}

layer_state_t layer_state_set_user(layer_state_t state) {
    writePin(B0, !layer_state_cmp(state, 1));
    return state;
}
```

Reply:
Hi @drashna ! Thank you for the super fast review, the code looks cleaner, but sadly the behaviour is not the expected 😢 

The idea is, whenever the layer 1 is activated, switch the led indicator on, if not, it should be off.

I tried this but it doesn't work, as expected, I can try to refactor the current code, I'm pretty new to the QMK API so I did the best I could reading the docs, but I'm happy to try and improve it though 😄

Reply:
Okay, I wasn't sure about which why it should be, 


This may work properly, then:
```suggestion
void keyboard_post_init_user(void) {
    setPinInputHigh(B0);
    writePinLow(B0);
}
layer_state_t layer_state_set_user(layer_state_t state) {
    writePin(B0, layer_state_cmp(state, 1));
    return state;
}
```

Reply:
Hi @drashna , sadly this doesn't work either 😢 , I've tried changing it a bit and I ended up with a cleaner solution, but I have to rely on `setPinInputHigh` and `setPinInputLow` when the layer changes.

By using `writePin` I see the light turns on but is just momentarily, the idea is to have a similar behavior as the caps lock key, that the light keeps on until the caps lock is pressed again.

Also, I added some tap dance behaviour to the `i` keycap, so it triggers the second layer when double tapping

I hope this is good now 😄